src/main/clojure/jarl/eval.clj

```diff
@@ -20,14 +20,14 @@
     (do
       (log/infof "ArrayAppendStmt - <%s> is not a local var" array-index)
       (break state))
-    (let [val (state/get-value state value-index)]
-      (if (nil? val)
-        (do
-          (log/debugf "ArrayAppendStmt - value <%s> not present" value-index)
-          (break state))
+    (if (state/contains-value? state value-index)
+      (let [val (state/get-value state value-index)]
         (let [array (state/get-local state array-index)]
```

Comment:
Should these not be a single `let`?

Reply:
Hm, yes. I didn't look at the very next line too closely, it would seem 😄 .